libp2p/Host.cpp

```diff
@@ -486,15 +485,10 @@ std::unordered_map<Public, std::string> Host::pocHosts()
     };
 }
 
-void Host::registerCapability(std::shared_ptr<CapabilityFace> const& _cap)
+void Host::registerCapability(std::shared_ptr<CapabilityFace> _cap)
 {
-    registerCapability(_cap, _cap->name(), _cap->version());
-}
-
-void Host::registerCapability(
```

Comment:
I kept this on purpose.

The idea of having these two overloads is to allow one Capability implementation to handle two (or more) protocol versions. It was used previously to support both `eth/62` and `eth/63`, now I removed support for 62 for the sake of simplification, but it might be needed if at some point there's `eth/64` and we have to support it.
Also in general seems to be a useful feature of p2p library.
(though not properly tested I think)

Reply:
(that is, it's needed if there are clients in the network that support only older version, but we still want to talk to them)